build-script.py

```diff
@@ -315,28 +355,36 @@ def build(self, product_name, module_group_path=""):
 
 
 def verify_generated_files(gyb_exec, verbose):
-    user_generated_dir = os.path.join(
-        PACKAGE_DIR, "Sources", "SwiftSyntax", "gyb_generated"
+    user_swiftsyntax_generated_dir = os.path.join(
+        SWIFTSYNTAX_DIR, "gyb_generated"
+    )
+
+    user_swiftsyntaxbuilder_generated_dir = os.path.join(
+        SWIFTSYNTAXBUILDER_DIR, "gyb_generated"
     )
 
     self_generated_dir = tempfile.mkdtemp()
     generate_gyb_files(
         gyb_exec,
         verbose=verbose,
         add_source_locations=False,
-        destination=self_generated_dir,
+        swiftsyntax_destination=self_generated_dir,
+        swiftsyntaxbuilder_destination=self_generated_dir,
```

Comment:
I just checked and `self_generated_dir` needs to be different for `SwiftSyntax` and `SwiftSyntaxBuilder`. As a little background what `verify_generated_files` is for: You already noticed that we don’t only check in the `.swift.gyb` files but also the generated `.swift` files. This is because SwiftSyntax should be a standalone Swift package and users of SwiftSyntax wouldn't be able to generate the files themselves (because SwiftPM doesn’t allow source generation, because the user probably doesn't have `gyb` installed on their system and because they are missing the Python files in the main Swift repo). Now, we do of course want to make sure that the generated files actually match the `.gyb` templates and that you don’t forget to check them in. This is why, in CI we are (more or less) running 

```bash
./build-script.py --toolchain /path/to/toolchain/ --degyb-only --verify-generated-files
```

which will fail if `gyb`-generation produces different files than those checked into the repo. If you want, you can try it yourself: Modify a file in `gyb_generated` and run the above command.

Back to the code, I was fiddling around locally and would propose the following change if you like it.

```python
def verify_generated_files(gyb_exec, verbose):
    user_swiftsyntax_generated_dir = os.path.join(
        SWIFTSYNTAX_DIR, "gyb_generated"
    )

    user_swiftsyntaxbuilder_generated_dir = os.path.join(
        SWIFTSYNTAXBUILDER_DIR, "gyb_generated"
    )

    self_swiftsyntax_generated_dir = tempfile.mkdtemp()
    self_swiftsyntaxbuilder_generated_dir = tempfile.mkdtemp()
    generate_gyb_files(
        gyb_exec,
        verbose=verbose,
        add_source_locations=False,
        swiftsyntax_destination=self_swiftsyntax_generated_dir,
        swiftsyntaxbuilder_destination=self_swiftsyntaxbuilder_generated_dir,
    )

    check_generated_files_match(self_swiftsyntax_generated_dir, user_swiftsyntax_generated_dir)
    check_generated_files_match(self_swiftsyntaxbuilder_generated_dir, user_swiftsyntaxbuilder_generated_dir)

def check_generated_files_match(self_generated_dir, user_generated_dir):
    command = [
        "diff",
        "-r",
        "-x",
        ".*",  # Exclude dot files like .DS_Store
        "--context=0",
        self_generated_dir,
        user_generated_dir,
    ]
    check_call(command)
```